### DISCUSSION TITLE
Rules for locating github actions

### DISCUSSION THREAD

**Discussion by DennisHeimbigner** [community/community#27222] [Actions]
<p>I am trying to figure out the rules by which github locates<br>
workflows to show on the left of the Actions page. Specifically:</p>
<ol>
<li>
<p>Is it possible to have different sets of workflows on different branches? My experiments seem to indicate that only workflows on the master branch are used.</p>
</li>
<li>
<p>Is it the case that whenever I visit the actions page, github looks in .github/workflows and displays all the .yml files there as workflows? Again by experimentation, this does not appear to be the case because I can add a workflow via PR but it does not show up.</p>
</li>
<li>
<p>Does it matter that the repo in which I am working is a fork of another repo?</p>
</li>
</ol>

**Comment by darkRaspberry** [comment]
<p>I guess this is helpful <a href="https://github.com/features/actions" rel="noopener nofollow ugc">GitHub actions</a>.</p>
<ol>
<li>Yes it is possible to have a different set of workflow according to branch.</li>
<li>I can’t understand this. BUT You can add new workflow through PR and it will show in GitHub actions. If its not showing there. Then there might some other isssue.</li>
</ol>
<aside class="quote no-group" data-username="DennisHeimbigner" data-post="1" data-topic="147170">
<div class="title">
<div class="quote-controls"></div>
 DennisHeimbigner:</div>
<blockquote>
<p>Again by experimentation, this does not appear to be the case because I can add a workflow via PR but it does not show up.</p>
</blockquote>
</aside>
<ol start="3">
<li>I guess it doesn’t matter. I can use actions on any repo including fork.</li>
</ol>

**Comment by DennisHeimbigner** [comment]
<p>Thanks. I still see problems with workflows on non-master branches.</p>
<p>Also, I saw on stack exchange the claim that a workflow will not show up until its first activation. Is this correct? If so, it seems like a bad idea.</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="DennisHeimbigner" data-post="1" data-topic="147170">
<div class="title">
<div class="quote-controls"></div>
 DennisHeimbigner:</div>
<blockquote>
<p>Is it possible to have different sets of workflows on different branches? My experiments seem to indicate that only workflows on the master branch are used.</p>
</blockquote>
</aside>
<p>That depends on the event that’s supposed to trigger the workflow, key quote from the documentation (link below):</p>
<blockquote>
<ol>
<li>An event occurs on your repository, and the resulting event has an associated commit SHA and Git ref.</li>
<li>The <code>.github/workflows</code> directory in your repository is searched for workflow files at the associated commit SHA or Git ref. The workflow files must be present in that commit SHA or Git ref to be considered.</li>
</ol>
</blockquote>
<p>If you check the event descriptions on that page the <code>GITHUB_REF</code> describes which branch (or other ref) the event uses.</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="20" height="20">

<h3><a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows" target="_blank" rel="noopener">Events that trigger workflows - GitHub Docs</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by DennisHeimbigner** [comment]
<p>Thanks for the documentation pointer.</p>
